packages/core/src/app/customer/GuestForm.tsx

```diff
@@ -38,7 +49,213 @@ const GuestForm: FunctionComponent<GuestFormProps & WithLanguageProps & FormikPr
     onShowLogin,
     privacyPolicyUrl,
     requiresMarketingConsent,
+    isStripeLinkEnabled,
+    updateStripeLinkAuthenticated,
+    checkoutStep,
+    deinitialize,
+    initialize,
+    onContinueAsGuest,
+    isStripeLinkAuthenticated
+
 }) => {
+    const [continueAsAGuestButton, setContinueAsAGuestButton] = useState(true);
+    const [emailValue, setEmailValue] = useState('');
+    const [isStripeLoading, setIsStripeLoading] = useState(true);
+    const [isNewAuth, setIsNewAuth] = useState(false);
+    const handleOnClickSubmitButton = () => {
+        onContinueAsGuest({
+            email: emailValue,
+            shouldSubscribe: false,
+        });
+    };
+    const buttonText = isStripeLinkAuthenticated && !isNewAuth? 'customer.continue_as_stripe_customer_action' : continueAsGuestButtonLabelId;
```

Comment:
What if we were to add another `StripeLink`-like product to checkout,  current `buttonText`. `setEmailCallback` and style solutions would make things not sustainable.

We can discuss a proper `GuestFormProps` interface to support `StripeLink` on the following meeting calls.
cc @animesh1987

Reply:
Can we move all the stripe related logic to its own component? `<StripeGuestForm />` with the props that are only required by that component.